Common/Securities/Option/StrategyMatcher/OptionStrategyDefinitionMatch.cs

```diff
@@ -0,0 +1,190 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace QuantConnect.Securities.Option.StrategyMatcher
+{
+    /// <summary>
+    /// Defines a match of <see cref="OptionPosition"/> to a <see cref="OptionStrategyDefinition"/>
+    /// </summary>
+    public class OptionStrategyDefinitionMatch : IEquatable<OptionStrategyDefinitionMatch>
+    {
+        /// <summary>
+        /// The <see cref="OptionStrategyDefinition"/> matched
+        /// </summary>
+        public OptionStrategyDefinition Definition { get; }
+
+        /// <summary>
+        /// The number of times the definition is able to match the available positions.
+        /// Since definitions are formed at the 'unit' level, such as having 1 contract,
+        /// the multiplier defines how many times the definition matched. This multiplier
+        /// is used to scale the quantity defined in each leg definition when creating the
+        /// <see cref="OptionStrategy"/> objects.
+        /// </summary>
+        public int Multiplier { get; }
+
+        /// <summary>
+        /// The <see cref="OptionStrategyLegDefinitionMatch"/> instances matched to the definition.
+        /// </summary>
+        public IReadOnlyList<OptionStrategyLegDefinitionMatch> Legs { get; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="OptionStrategyDefinitionMatch"/> class
+        /// </summary>
+        public OptionStrategyDefinitionMatch(
+            OptionStrategyDefinition definition,
+            IReadOnlyList<OptionStrategyLegDefinitionMatch> legs,
+            int multiplier
+            )
+        {
+            Legs = legs;
+            Multiplier = multiplier;
+            Definition = definition;
+        }
+
+        /// <summary>
+        /// Deducts the matched positions from the specified <paramref name="positions"/>
+        /// </summary>
+        public OptionPositionCollection RemoveFrom(OptionPositionCollection positions)
+        {
+            return positions.RemoveRange(Legs.Select(leg => leg.Position));
+        }
+
+        /// <summary>
+        /// Creates the <see cref="OptionStrategy"/> instance this match represents
+        /// </summary>
+        public OptionStrategy CreateStrategy()
+        {
+            var legs = Legs.Select(leg => leg.CreateOptionStrategyLeg());
+            var strategy = new OptionStrategy {Name = Definition.Name, Underlying = Legs[0].Position.Underlying};
+            foreach (var leg in legs)
+            {
+                leg.Invoke(strategy.UnderlyingLegs.Add, strategy.OptionLegs.Add);
+            }
+
+            return strategy;
```

Comment:
Wouldn't be able to use the _definition_ directly, but perhaps the _definition match_ as it contains the definition name and the multiplier. The motivation behind pushing everything into the `OptionStrategy` format was standardization since that's what is accepted by the `QCAlgorithm` api. The margin models are going to need the strategy name and the individual positions/multiplier. We'll use the name to do a lookup in a table to get the function to evaluate the margin. Individual positions, specifically strikes, are required by some of these margin computation functions.